apps/storefront/src/components/HeadlessController/index.test.tsx

```diff
@@ -0,0 +1,150 @@
+    },
+    updatedAt: faker.date.anytime().toString(),
+    approvedFlag: faker.datatype.boolean(),
+    isOwner: faker.datatype.boolean(),
+    companyInfo: {
+      companyId: faker.string.numeric(),
+      companyName: faker.company.name(),
+      companyAddress: faker.location.streetAddress(),
+      companyCountry: faker.location.country(),
+      companyState: faker.location.state(),
+      companyCity: faker.location.city(),
+      companyZipCode: faker.location.zipCode(),
+      phoneNumber: faker.string.numeric(),
+      bcId: faker.string.numeric(),
+    },
+  },
+}));
+
+describe('HeadlessController shopping lists utils', () => {
+  it('getLists retrieves B2B shopping lists', async () => {
+    const getShoppingLists = vi.fn();
+
+    when(getShoppingLists)
+      .calledWith(stringContainingAll('first: 50', 'offset: 0'))
+      .thenReturn({
+        data: {
+          shoppingLists: {
+            totalCount: 1,
+            pageInfo: {
+              hasNextPage: false,
+              hasPreviousPage: false,
+            },
+            edges: [
+              buildShoppingListsNodeWith({
+                node: {
+                  id: 123,
+                },
+              }),
+            ],
+          },
+        },
+      });
+
+    server.use(
+      graphql.query('B2BCustomerShoppingLists', ({ query }) =>
+        HttpResponse.json(getShoppingLists(query)),
+      ),
+    );
+
+    const mockSetOpenPage = vi.fn();
+    renderWithProviders(<HeadlessController setOpenPage={mockSetOpenPage} />, {
+      preloadedState: {
+        company: buildCompanyStateWith({
+          customer: {
+            role: 3,
+          },
+        }),
+      },
+    });
+
+    const data = await window.b2b.utils.shoppingList.getLists();
+
+    expect(data).toBeDefined();
+    expect(data).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
+          id: 123,
+        }),
+      ]),
+    );
+  });
+
+  it('getLists retrieves B2C shopping lists', async () => {
+    const getShoppingLists = vi.fn();
+
+    when(getShoppingLists)
+      .calledWith(stringContainingAll('first: 50', 'offset: 0'))
+      .thenReturn({
+        data: {
+          customerShoppingLists: {
+            totalCount: 1,
+            pageInfo: {
+              hasNextPage: false,
+              hasPreviousPage: false,
+            },
+            edges: [
+              buildShoppingListsNodeWith({
+                node: {
+                  id: 123,
+                },
+              }),
+            ],
+          },
+        },
+      });
+
+    server.use(
+      graphql.query('CustomerShoppingLists', ({ query }) =>
+        HttpResponse.json(getShoppingLists(query)),
+      ),
+    );
+
+    const mockSetOpenPage = vi.fn();
+    renderWithProviders(<HeadlessController setOpenPage={mockSetOpenPage} />);
+    const data = await window.b2b.utils.shoppingList.getLists();
+
+    expect(data).toBeDefined();
+    expect(data).toEqual(
```

Comment:
if not defined, the `toEqual` will fail.

```suggestion
    expect(data).toEqual(
```